src/wallet/rpcwallet.cpp

```diff
@@ -3990,10 +3990,19 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
     if (!EnsureWalletIsAvailable(fHelp))
         return NullUniValue;
 
+    if (!fEnableZGetBalance)
+        throw runtime_error(
+            "z_getbalance is DEPRECATED and will be removed in a future release\n\n"
+            "Use z_getbalanceforaccount, z_getbalanceforviewingkey, or getbalance (for\n"
+            "legacy transparent keys) instead, or restart with `-allowdeprecated=z_getbalance`\n"
```

Comment:
`getbalance` doesn't query by key or address.
```suggestion
            "legacy transparent balance) instead, or restart with `-allowdeprecated=z_getbalance`\n"
```